docs/proposals/cluster-discovery-api.md

```diff
@@ -0,0 +1,91 @@
+# Cluster Discovery API
+
+Author: [@mikedanese](https://github.com/mikedanese)
+
+#### Abstract
+
+This document describes a mechanism for a kubelet to discover, from a trusted source, the
```

Comment:
> Only kubelet? Why not other components?

I think this works for many client usecases, not just kubelet. Will change

> Where would etcd fit into this?

Only the apiserver needs to know where etcd. Bootstrapping the etcd client in the apiserver is a separate problem then the one this proposal tries to solve. Here, we assume that the control-plane is up.